src/org/exist/Indexer.java

```diff
@@ -398,7 +398,6 @@ private void processText(ElementImpl last, ProcessTextParent ptp) {
     @Override
     public void endElement(final String namespace, final String name, final String qname) {
         final ElementImpl last = stack.peek();
-        if (last.getNodeName().equals(qname)) {
```

Comment:
@wolfgangmm So this check was incorrect somehow? But is it safe to actually remove it?

Reply:
@adamretter The test doesn't properly respect namespaces. But I see no reason why the node name passed to endElement would ever be different than the one startElement received on the same hierarchical level - unless the XML is actually invalid. In the latter case the XML parser would have complained during the validation step though. We could certainly fix the check, but in my opinion it is completely unnecessary and smells like old code left behind.

I did not see any side effects despite having tested this PR quite extensively. The 0-byte documents this bug produces are a major showstopper for many users and have been reported occasionally since at least half a year.

Reply:
Okay sounds cool